sktime/transformations/series/summarize.py

```diff
@@ -3,38 +3,219 @@
+        to summarize each column of the dataset.
+    axis : int, default=0
+        Axis to apply the summary function along. Only used for multivariate data.
+
+    Raises
+    ------
+    ValueError :
+        `series_or_df` must be pd.Series or pd>DataFrame.
+    ValueError :
+        `summary_function must be one of ("mean", "min", "max", "median", "sum",
+        "skew", "kurtosis", "var", "std", "sem", "nunique", "count").
+
+    Returns
+    -------
+    summary_value : scalar or pd.Series
+        If `series_or_df` is univariate then a scalar is returned. Otherwise,
+        a pd.Series is returned.
+    """
+    if isinstance(series_or_df, pd.Series):
+        is_series = True
+    elif isinstance(series_or_df, pd.DataFrame):
+        is_series = False
+    else:
+        raise ValueError("`series_or_df` must be pd.Series or pd>DataFrame.")
+
+    if summary_function == "mean":
+        summary_value = series_or_df.mean(axis=axis)
+    elif summary_function == "min":
+        summary_value = series_or_df.min(axis=axis)
+    elif summary_function == "max":
+        summary_value = series_or_df.max(axis=axis)
+    elif summary_function == "median":
+        summary_value = series_or_df.median(axis=axis)
+    elif summary_function == "sum":
+        summary_value = series_or_df.sum(axis=axis)
+    elif summary_function == "skew":
+        summary_value = series_or_df.skew(axis=axis)
+    elif summary_function == "kurtosis":
+        summary_value = series_or_df.kurtosis(axis=axis)
+    elif summary_function == "std":
+        summary_value = series_or_df.std(axis=axis)
+    elif summary_function == "var":
+        summary_value = series_or_df.var(axis=axis)
+    elif summary_function == "mad":
+        summary_value = series_or_df.mad(axis=axis)
+    elif summary_function == "sem":
+        summary_value = series_or_df.sem(axis=axis)
+    elif summary_function == "nunique":
+        if is_series:
+            summary_value = series_or_df.nunique()
+        else:
+            summary_value = series_or_df.nunique(axis=axis)
+    elif summary_function == "count":
+        if is_series:
+            summary_value = series_or_df.count()
+        else:
+            summary_value = series_or_df.count(axis=axis)
+    else:
+        raise ValueError(f"`summary_function must be one of {ALLOWED_SUM_FUNCS}.")
+
+    # For univariate data in DataFrame return scalar like when input is pd.Series
+    if is_series is False and series_or_df.shape[1] == 1:
+        summary_value = summary_value.iloc[0]
+
+    return summary_value
+
+
+class SummaryTransformer(_SeriesToPrimitivesTransformer):
+    """Calculate summary value of a time series.
+
+    Calculates scalar value when applied to a :term:`univariate time series`.
+    If input is :term:`multivariate time series` then Scalar values for each
+    column and returned as elements of a pd.Series.
+
+    Parameters
+    ----------
+    summary_function : str, default="mean"
```

Comment:
I think this should allow to choose multiple, or all. Otherwise it's clunky to use.